src/AutoHashEqualsCached.jl

```diff
@@ -2,10 +2,27 @@
 
 module AutoHashEqualsCached
 
-using Rematch
+using Pkg
 
 export @auto_hash_equals_cached, @auto_hash_equals
 
+pkgversion(m::Module) = VersionNumber(Pkg.TOML.parsefile(joinpath(dirname(string(first(methods(m.eval)).file)), "..", "Project.toml"))["version"])
```

Comment:
```suggestion
pkgversion(m::Module) = VersionNumber(Pkg.TOML.parsefile(joinpath(pkgdir(m), "Project.toml"))["version"])
```

Reply:
`pkgdir` doesn't do what we need.  It would "Return the root directory of the package that imported module `m`, or nothing if `m` was not imported from a package".  But we want the root directory of the package *containing* `m`, not the package that *imported* `m`.  A better implementation would just call `[Base.pkgversion](https://docs.julialang.org/en/v1/base/base/#Base.pkgversion-Tuple{Module})`, but that requires Julia 1.9.

Reply:
I'm not entirely convinced the docstring is correct there. The example shows it doing exactly what we want, and my tests at the repl have all returned the expected directory. Have you seen cases where this doesn't do what you expect?

Reply:
Will change per your suggestion... and also fix the doc here: https://github.com/JuliaLang/julia/pull/50255